pandacommon/kafkapublisher/KafkaPublisher.py

```diff
@@ -6,25 +6,34 @@
 
 import json
 import hashlib
+import socket
 from confluent_kafka import Producer
 from pandacommon.pandalogger import logger_utils
-from . import kafka_config as config
+from pandacommon.commonconfig import common_config
 
 class KafkaPublisher:
     def __init__(self):
+        kafka_config = common_config.get('kafka')
         self.producer = Producer({
-                'bootstrap.servers': config.BOOTSTRAP_SERVERS,
-                'group.id': config.GROUP_ID,
-                'ssl.ca.location': config.CACERTS,
+                'bootstrap.servers': self.get_bootstrap_servers(kafka_config['kafka_cluster']),
+                'group.id': kafka_config['group_id'],
+                'ssl.ca.location': kafka_config['cacerts'],
                 'security.protocol': 'SASL_SSL',
-                'sasl.kerberos.keytab': config.KEYTAB,
+                'sasl.kerberos.keytab': kafka_config['keytab'],
                 'auto.offset.reset': 'latest',
                 'enable.auto.offset.store': True,
-                'sasl.kerberos.principal': config.PRINCIPAL,
+                'sasl.kerberos.principal': kafka_config['principal'],
                 'log_level': 0
         })
         self.logger = logger_utils.setup_logger()
 
+    def get_bootstrap_servers(self, cluster):
+        return ",".join(
+            map(lambda x: x + ":9093",
+                sorted([(socket.gethostbyaddr(i))[0] for i in (socket.gethostbyname_ex(cluster + ".cern.ch"))[2]])
```

Comment:
Can you change .cern.ch as a config parameter? I suppose bootstrap.servers is a parameter independent of the auth method and thus needs to be set also when using non-CERN kafka.

Reply:
done, thanks!